### ISSUE TITLE
Integer Division is confuse

### ISSUE THREAD

**Issue by lspigariol** [uqbar-project/wollok#924]
now, between integers / returns another integer, but common people expect real value

for example 3 / 10 returns 0
 
If the user wanted that kind of behaviour, he could use .div() which always returns integer

**Comment by mmatos** [comment]
Agree, not only is confusing, also if you know that's how it behaves, having to add .0 to all numbers around the program / tests that are expected to return a float when being divided is a pain.

**Comment by npasserini** [comment]
Agree.

I also see here a more profound discussion pending. I ask myself if we want to have two kinds of numbers, why not only numbers? Obviously this is not a change to introduce now, but I would like that we talk about it.
I understand that it will be different from most (or all!) languages, but do we want to teach that in our OOP subjects? Do we want to talk about this differences?

I am not sure if I can think of a domain necessity for integer numbers, separated from real numbers. I think they only exist because of underlying hardware problems, legacy and performance issues, which we could avoid completely.

**Comment by asanzo** [comment]
Agree. Let the division function as is intuitively expected. 3/2 should return 1.5
I've just had the same issue programming in Python a couple of days ago. 

Aaaand right now a student has complained about this in Wollok.

Regarding domain necessity, I also don't know. I can only think in two cases where differentiation among number types is useful when teaching:

1. How the number is written and seen (I don't want to write .0 everywhere, and I'd like the integer values to be seen without .0)
1. Some use the internal representation of a language to teach some concepts (e.g. collections or numbers hierarchy when explaining inheritance, numbers inner representation when explaining classes). 

I don't use number 2 as it focus on the _how_ things are done rather than _what_ do they represent. So I think I wouldn't mind that everything was just "Number".

**Comment by lspigariol** [comment]
like only numbers, whith methods to represent on disctincts formats. 

on that road,  
5/2 returns 2.5
4/2 returns 2 (or 2.0)

**Comment by npasserini** [comment]
If we have only numbers, I do not see why we cant have the number 2
printing as 2 instead of 2.0.

**Comment by lspigariol** [comment]
Agree.
2.0 is as futile as 02

**Comment by fdodino** [comment]
Didn't understand what [If we have only numbers, I do not see why we cant have the number 2
printing as 2 instead of 2.0] means.

I'll add some tests:
Expectations
3 / 10 should return 0.3
5/2 should return 2.5
4/2 should return 2

**Comment by fdodino** [comment]
Added also that 4 / 2.0 should return 2

**Comment by fdodino** [comment]
ArithmeticTest had
#["3 / 2", 1],
strange!
Fixed it.

**Comment by npasserini** [comment]
Two important things

1) Are we going to include this fix in version 1.5.x? Shouldn't we wait for
1.6/2.0?
2) If we make / message uniform (i.e. 5/3 ==5.0/3.0), I think we should
have a separate operator for integer division/remainder, for example

5//3 = 1 and 5%3 = 2
while
3//2 = 1 and 3%2 = 1

I would not know what to do about 2.5//2 nor 2.5%2

Also, I am not sure about symbols // and % maybe we should choose others,
or even use named messages, such as integerDivision and remainder ?

**Comment by lspigariol** [comment]
about division returns float, i prefer 1.5.x
unify numbers, can wait.
 
`2.5%2` returns 0
`3.5%2` returns 1

i thik % firt truncate and after calcule module
// dont exists, the message is .div(), but the expectable behavior is similar

Actually, is not practical de `x.div(y)` sintaxis. would more friendly `x div y`
for standarize, can do `5 mod 2` instead of `5%2`

**Comment by fdodino** [comment]
sorry, agree with Lucas except x div y, which breaks Wollok syntax for symbols and operators.
